packages/xforms-engine/src/instance/text/TextRange.ts

```diff
@@ -21,9 +33,19 @@ export class TextRange<Role extends TextRole, Origin extends TextOrigin>
 		return this.chunks.map((chunk) => chunk.asString).join('');
 	}
 
+	get image(): Promise<MediaResource> | null {
+		if (this.media?.image == null || !isEngineXPathEvaluator(this.evaluator)) {
+			return null;
+		}
+
+		return (this.evaluator.rootNode as PrimaryInstance).loadMediaResources(this.media.image);
```

Comment:
The type guard `isEngineXPathEvaluator` verifies that evaluator is `EngineXPathEvaluator`, which `rootNode` is already a `PrimaryInstance`. However, the type static check is confused because XPath defines `rootNode` as some other general type and does not explicitly define a `PrimaryInstance`. This cast is safe.